scripts/end2end_test.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/sh
+
+set -x # debugging
+
+DOCKER_COMPOSE_VERSION="1.8.1"
+DOCKER_COMPOSE_EXEC="/tmp/docker-compose"
+DOCKER_COMPOSE_FILE="docker/docker-compose.yml"
+
+curl -L "https://github.com/docker/compose/releases/download/${DOCKER_COMPOSE_VERSION}/docker-compose-Linux-x86_64" > ${DOCKER_COMPOSE_EXEC}
+chmod +x ${DOCKER_COMPOSE_EXEC}
+
+# start all docker containers
+${DOCKER_COMPOSE_EXEC} -f ${DOCKER_COMPOSE_FILE} up -d
+
+# wait for carbon input before sending data
+export WAIT_HOSTS="127.0.0.1:2003"
+export WAIT_TIMEOUT=120
+export METRICS_PER_SECOND=1000
+scripts/wait_for_endpoint.sh scripts/generate_test_data.sh start
+
+# give fakemetrics some warmup time
+sleep 30
+
+# verify the metrics have arrived in graphite and keep exit status
+scripts/verify_metrics_received.py 127.0.0.1 8080 10 ${METRICS_PER_SECOND}
+RESULT=${?}
```

Comment:
Actually that usage of ${} is just a habit of mine because it also doesn't have a real disadvantage, and it prevents me from mistakenly omitting the {} when they would have been necessary.
I can remove them at the places where they aren't necessary.